invalid memory address or nil pointer dereference

启动没一会就报这个错，可以帮看一下吗，数据库版本mysql8.0.25

{"level":"info","msg":"DSN is root:123456@tcp(192.168.3.8:3306)/?interpolateParams=true\u0026timeout=5s\u0026readTimeout=5s\u0026writeTimeout=5s\u0026parseTime=true\u0026collation=utf8mb4_general_ci\u0026loc=Local","time":"2023-05-06T19:42:09+08:00"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xbc2e86]

goroutine 1 [running]:
reflect.mapiternext(0xc000041320?)
        /app/go/src/runtime/map.go:1380 +0x19
github.com/modern-go/reflect2.(*UnsafeMapIterator).UnsafeNext(0xc000a27218?)
        /root/go/pkg/mod/github.com/modern-go/reflect2@v1.0.1/unsafe_map.go:136 +0x32
github.com/json-iterator/go.(*sortKeysMapEncoder).Encode(0xc0001d07e0, 0xc000812178, 0xc00031e180)
        /root/go/pkg/mod/github.com/json-iterator/go@v1.1.11/reflect_map.go:297 +0x315
github.com/json-iterator/go.(*structFieldEncoder).Encode(0xc0001d0db0, 0x1d67059?, 0xc00031e180)
        /root/go/pkg/mod/github.com/json-iterator/go@v1.1.11/reflect_struct_encoder.go:110 +0x56
github.com/json-iterator/go.(*structEncoder).Encode(0xc0001d1230, 0x2408f80?, 0xc00031e180)
        /root/go/pkg/mod/github.com/json-iterator/go@v1.1.11/reflect_struct_encoder.go:158 +0x488
github.com/json-iterator/go.(*Stream).WriteVal(0xc00031e180, {0x1f53c20, 0xc000812140})
        /root/go/pkg/mod/github.com/json-iterator/go@v1.1.11/reflect.go:98 +0x158
github.com/json-iterator/go.(*frozenConfig).MarshalToString(0xc00028f4a0, {0x1f53c20, 0xc000812140})
        /root/go/pkg/mod/github.com/json-iterator/go@v1.1.11/config.go:289 +0xa6
github.com/moiot/gravity/pkg/inputs/mysqlbatch.EncodeBatchPositionValue({0x1f53c20?, 0xc000812100?})
        /app/gravity/pkg/inputs/mysqlbatch/position_value.go:242 +0x1b5
github.com/moiot/gravity/pkg/position_cache.(*defaultPositionCache).encodePositionValueString(0x0?, 0x0?)
        /app/gravity/pkg/position_cache/cache.go:245 +0x28
github.com/moiot/gravity/pkg/position_cache.(*defaultPositionCache).Put(0xc000490870, {{{0x0, 0x0}, {0xc000616f10, 0xe}, {0x20c7ee5, 0x5}, {0xc10dad24772559bc, 0x1b2f4eb6, 0x3617080}}, ...})
        /app/gravity/pkg/position_cache/cache.go:106 +0x135
github.com/moiot/gravity/pkg/inputs/mysqlbatch.SetupInitialPosition({0x2402400, 0xc000490870}, 0x7f00b822c658?)
        /app/gravity/pkg/inputs/mysqlbatch/position_value.go:327 +0x2bd
github.com/moiot/gravity/pkg/inputs/mysqlbatch.(*mysqlBatchInputPlugin).NewPositionCache(0xc0001e2b40)
        /app/gravity/pkg/inputs/mysqlbatch/input.go:174 +0x10b
github.com/moiot/gravity/pkg/inputs/helper.(*TwoStageInputPlugin).NewPositionCache(0xc000612140)
        /app/gravity/pkg/inputs/helper/two_stage_input.go:51 +0xa4
github.com/moiot/gravity/pkg/inputs/mysql.(*input).NewPositionCache(0xc000142780?)
        /app/gravity/pkg/inputs/mysql/input.go:76 +0x35
github.com/moiot/gravity/pkg/app.NewServer({{0xc000142780, 0xe}, {0xc0006167d0, 0x8}, {0x20c60f5, 0x3}, {{0xc0006167f0, 0x5}, {0xc000616820, 0xb}, ...}, ...})
        /app/gravity/pkg/app/server.go:121 +0xb3
main.main()
        /app/gravity/cmd/gravity/main.go:83 +0x5b8

--- a/go.mod
+++ b/go.mod
@@ -16,7 +16,7 @@ require (
 	github.com/hashicorp/go-plugin v0.0.0-20190220160451-3f118e8ee104
 	github.com/hashicorp/yamux v0.0.0-20181012175058-2f1d1f20f75d // indirect
 	github.com/jinzhu/gorm v1.9.12
-	github.com/json-iterator/go v1.1.11
+	github.com/json-iterator/go v1.1.12
 	github.com/juju/errors v0.0.0-20170703010042-c7d06af17c68
 	github.com/juju/loggo v0.0.0-20190212223446-d976af380377 // indirect
 	github.com/juju/testing v0.0.0-20180920084828-472a3e8b2073 // indirect

